### ISSUE TITLE
Import from file to provision alerting

### ISSUE THREAD

**Issue by jtheory** [grafana/grafana#48623]
This could also be a zip file with many files in it.

**Comment by Geigerkind** [comment]
Hi @jtheory ,
do I read this correctly that there is currently no way to provision alerting through simple files?
I am struggeling to find documentation on how to provision my alert rules for my kubernetes cluster but I just cant find any.
Is there any workaround that I can currently use?

**Comment by sarasensible** [comment]
@Geigerkind  there is a workaround but there is also now a supported alert rule file-based provisioning method that is available through Grafana's 9.1 beta release.

Workaround via the API (I had to port-forward to get this to work since we are using SSO in front of Grafana. Replace localhost:3000 with your Grafana site url):
1. `curl -X GET -H "Authorization: Bearer $GRAFANA_TOKEN" -H "Content-type: application/json" "http://localhost:3000/api/ruler/grafana/api/v1/rules/" | jq > grafana_alerts.json`
2. Bash script sourced from https://community.grafana.com/t/ngalert-grafana-8-alert-feature-how-to-export-import-alerts-as-yml-json/51677/22?u=sarasensible
```
#!/bin/sh

ALERTS_JSON_PATH=./grafana_alerts_prod.json
NUMBER_OF_ALERTS=$(jq -c ".folder-name | length" ${ALERTS_JSON_PATH})

for ((i=0; i<NUMBER_OF_ALERTS; i++)); do
  ALERT_OBJECT=$(jq -c --arg i "$i" '.folder-name[($i | tonumber)] | del(.rules[].grafana_alert.uid)' ${ALERTS_JSON_PATH})
  ALERT_NAME=$(jq -c --arg i "$i" '.folder-name[($i | tonumber)].name' ${ALERTS_JSON_PATH})
  echo "Creating ${ALERT_NAME}...\n"
  echo $ALERT_OBJECT
	curl -X POST \
		-H "Authorization: Bearer ${GRAFANA_TOKEN}" \
		-H "Content-type: application/json" \
		'http://localhost:3000/api/ruler/grafana/api/v1/rules/folder-name' \
		-d "${ALERT_OBJECT}"
	echo "\n"
done
```

Supported (BETA) file based provisioning - the configs I used are available in https://github.com/grafana/grafana/discussions/53490 .

**Comment by Geigerkind** [comment]
@sarasensible Those are wonderful news. I also thought about using this API, but this seemed to me a little bit too hacky to implement into my GCP GKE. We downgraded to version 7.2 which hat a working alerting for timelines. Im looking forward to upgrading to the new version, once its out of beta. Thanks for notifying me!
